### PR TITLE
added cub_dataloader

### PR DESCRIPTION
**Pull request by mridulk97**

### CODE REVIEW

File: data/cub_list_data_files.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import glob
+
+train = []
+for dir in glob.glob('fastscratch/mridul/cub_190_split_resized/official/CUB_200_2011/train_256/*'):
+    train.append(glob.glob(str(dir)+'/*'))
```
Comment (elhamod) [inline data/cub_list_data_files.py]: Any file paths or strings that may change from user to user should be added as parameters, not hardcoded.

**Review by elhamod — CHANGES_REQUESTED** [review]

File: data/cub_list_data_files.py (hunk 1 above)
Comment (mridulk97) [reply data/cub_list_data_files.py]: done

File: taming/data/cub_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from torch.utils.data import Dataset
+from taming.data.base import ImagePaths
```
Comment (elhamod) [inline taming/data/cub_dataloader.py]: Looking at this again, I don not see a good reason for reason for creating a duplicate of the Custom dataset objects. either use inheritance or just use same objects

**Review by elhamod — CHANGES_REQUESTED** [review]

File: taming/data/image_squarer.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        if smaller_dimension == 1:
+            img = transforms.functional.resize(img, (new_smaller_dimension, imageDimension))
+        else:
+            img = transforms.functional.resize(img, (imageDimension, new_smaller_dimension))
+
+        diff = imageDimension - new_smaller_dimension
+        pad_1 = int(diff/2)
+        pad_2 = diff - pad_1
+
+        # # imagenet mean
```
Comment (elhamod) [inline taming/data/image_squarer.py]: Please make this a flag.

File: taming/data/create_data_txt_files.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import os
+import glob
```
Comment (elhamod) [inline taming/data/create_data_txt_files.py]: This file is not needed. We already have this shell command: /home/elhamod/projects/taming-transformers/analysis/create_file_lists.sh :)

File: taming/data/annotate_cub_dataset_mask.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        new_folder = os.path.join(args.output_dir, species)
+        if not os.path.isdir(new_folder):
+            os.makedirs(new_folder)
+        for image_path in sorted(glob.glob(folder+'/*')):
+            image_file = image_path.split('/')[-1]
+            segmentation_path = os.path.join(args.segmentation_base_dir, species, image_file)
+            segmentation_path = segmentation_path.replace('jpg', 'png')
+            try:
+                
+                image = PIL.Image.open(image_path)
```
Comment (elhamod) [inline taming/data/annotate_cub_dataset_mask.py:31]: Maybe we should wait till you figure out what the gray areas in the mask are?

**Review by elhamod — CHANGES_REQUESTED** [review]

File: taming/data/annotate_cub_dataset_mask.py (hunk 5 above)
Comment (mridulk97) [reply taming/data/annotate_cub_dataset_mask.py:31]: I tried looking for it, but there is no documentation for it anywhere about the grey areas.

File: taming/data/image_squarer.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        for image_path in sorted(glob.glob(folder+'/*')):
+            image_file = image_path.split('/')[-1]
+            image = PIL.Image.open(image_path)
+            try:
+                squared_image = MakeSquared(image, padding=args.padding)
+                new_image_path = os.path.join(new_folder, image_file)
+                squared_image.save(new_image_path)
+            except Exception as error:
+                print(image_path)
+                print(error)
```
Comment (elhamod) [inline taming/data/image_squarer.py:66]: test.ipynb is empty. please remove it.

**Review by elhamod — APPROVED** [review]
